cuqipy_pytorch/distribution.py

```diff
@@ -395,3 +395,109 @@ def gradient(self, v1, v2=None):
             Q = self(v2).logpdf(v1) # Forward pass
             Q.backward()            # Backward pass
             return v2.grad
+
+class Laplace(Distribution):
+
+    def __init__(self, location, scale, is_symmetric=True, **kwargs):
+        super().__init__(is_symmetric=is_symmetric, **kwargs) 
+        self.location = location
+        self.scale = scale
+
+    @property
+    def location(self):
+        return self._location
+
+    @location.setter
+    def location(self, value):
+        self._location = value
+        self._set_dist()
+    
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, value):
+        self._scale = value
+        self._set_dist()
+
+    def _set_dist(self):
+        """ Set the pytorch distribution if both values are specified """
+        if hasattr(self, '_location') and hasattr(self, '_scale'):
+
+            # Define location value
+            loc = self.location
+            if isinstance(loc, numbers.Number):
+                loc = loc*torch.ones(self.dim)
+            if isinstance(loc, np.ndarray):
+                loc = torch.tensor(loc)
+
+            # Define scale value
+            scale = self._scale
+            if isinstance(scale, numbers.Number):
+                scale = scale*torch.ones(self.dim)
+            if isinstance(scale, np.ndarray):
+                scale = torch.tensor(scale)
+
+            # If both are tensors we create dist
+            if torch.is_tensor(loc) and torch.is_tensor(scale):
```

Comment:
What should happen in the else clause? should we add an error?

Reply:
We should not. Nothing should happen e.g. if its a lambda function